src/pages/index.js

```diff
@@ -1,109 +1,212 @@
+        api
+          .addLike(data._id)
+          .then((data) => {
+            card.updateLikes(data.isLiked);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      }
+    },
+    handleDeleteCard: (cardId) => {
+      deletePopup.open();
+      deletePopup.setSubmitAction(() => {
+        deletePopup.renderLoading(true);
+        api
+          .deleteCard(cardId)
+          .then(() => {
+            card.deleteClick();
+            deletePopup.close();
+          })
+          .catch((err) => {
+            console.log(err);
+          })
+          .finally(() => {
+            deletePopup.renderLoading(false);
+          });
+      });
+    },
+  });
+  return card.generateCard();
 }
 
-//PopUpWithForm
-const profileEditPopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleEditProfileSubmit
+const addFormValidator = new FormValidator(addModalForm, formValidationConfig);
+addFormValidator.enableValidation();
+
+const profileFormValidator = new FormValidator(
+  profileEditForm,
+  formValidationConfig
 );
-profileEditPopup.setEventListeners();
-profileEditButton.addEventListener("click", () => {
-  handleProfileEditClick();
-});
+profileFormValidator.enableValidation();
 
-function handleProfileEditClick() {
-  const user = userInfo.getUserInfo();
-  userNameInput.value = user.name;
-  userDescriptionInput.value = user.job;
-  profileEditPopup.open();
-}
+const aviFormValidator = new FormValidator(aviForm, formValidationConfig);
+aviFormValidator.enableValidation();
 
-function handleEditProfileSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues);
-  profileEditPopup.close();
-}
+const deletePopup = new PopupWithConfirmation("#delete-popup");
+deletePopup.setEventListeners();
 
-// Add Card Modal
-const addCardPopup = new PopupWithForm("#new-card-modal", handleAddCardSubmit);
-addCardPopup.setEventListeners();
-addCardButton.addEventListener("click", () => {
-  addCardFormValidator.resetValidation();
-  addCardPopup.open();
+const editFormModal = new PopupWithForm("#modal", (inputValues) => {
+  editFormModal.renderLoading(true);
+  api
+    .editProfile(inputValues)
+    .then(() => {
+      userInfoEl.setUserInfo({
+        name: inputValues.name,
+        job: inputValues.description,
+      });
+      editFormModal.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      editFormModal.renderLoading(false);
+    });
 });
+editFormModal.setEventListeners();
 
-function handleAddCardSubmit(inputValues) {
-  const newCard = createCard(inputValues);
-  section.addItem(newCard);
-  addCardPopup.close();
-}
+const addFormModal = new PopupWithForm("#add-popup", (inputValues) => {
+  addFormModal.renderLoading(true);
+  api
+    .addNewCard(inputValues)
+    .then((data) => {
+      const card = createCard(data);
+      sectionEl.addItem(card);
+      addFormModal.close();
+    })
+    .catch((err) => {
+      console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors